### PR TITLE
Move formatting analyzer to `Analyzers` layer

### PR DESCRIPTION
**Pull request by Youssef1313**
There were two versions of formatting analyzer. One in `Features` and the other is in `CodeStyle`. This PR makes a single analyzer in `Analyzers` layer, which is shared between both `CodeStyle` and `Features`.

### CODE REVIEW

File: src/Analyzers/Core/Analyzers/Formatting/FormattingAnalyzerHelper.cs [hunk 1]
```diff
@@ -8,12 +8,8 @@
 using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.Text;
 
-#if CODE_STYLE
 using Formatter = Microsoft.CodeAnalysis.Formatting.FormatterHelper;
 using FormattingProvider = Microsoft.CodeAnalysis.Formatting.ISyntaxFormatting;
-#else
-using FormattingProvider = Microsoft.CodeAnalysis.Host.HostWorkspaceServices;
-#endif
```
Comment (Youssef1313) [inline src/Analyzers/Core/Analyzers/Formatting/FormattingAnalyzerHelper.cs:16]: Previously, `Features` was using Mef and `CodeStyle` was using `ISyntaxFormatting`. Now both are using `ISyntaxFormatting`. @mavasani Can you confirm this is the right action?

Comment (mavasani) [reply src/Analyzers/Core/Analyzers/Formatting/FormattingAnalyzerHelper.cs:16]: I think it should be fine, @sharwell to confirm

File: src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixProvider.cs [hunk 2]
```diff
@@ -28,12 +26,14 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             foreach (var diagnostic in context.Diagnostics)
             {
+#pragma warning disable RS0005 // Do not use generic 'CodeAction.Create' to create 'CodeAction'
                 context.RegisterCodeFix(
```
Comment (Youssef1313) [inline src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixProvider.cs:30]: @mavasani The one that used to live in `Features` was inheriting `SyntaxEditorBasedCodeFixProvider`. While the one that lived in `CodeStyle` was inheriting `CodeFixProvider` directly. Does it matter what do I inherit when merging both in `Analyzers` layer?

Comment (mavasani) [reply src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixProvider.cs:30]: Moving to `SyntaxEditorBasedCodeFixProvider` is preferable if possible, and you don't see any breaking changes.

Comment (Youssef1313) [reply src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixProvider.cs:31]: @mavasani Done in last commit (5e4b835)

**Comment by mavasani** [comment]
@Youssef1313 There are merge conflicts

File: src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixProvider.cs [hunk 3]
```diff
@@ -28,12 +27,14 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             foreach (var diagnostic in context.Diagnostics)
             {
+#pragma warning disable RS0005 // Do not use generic 'CodeAction.Create' to create 'CodeAction'
```
Comment (mavasani) [inline src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixProvider.cs:30]: Unnecessary?

File: src/CodeStyle/Core/Analyzers/Microsoft.CodeAnalysis.CodeStyle.csproj [hunk 4]
```diff
@@ -26,7 +26,6 @@
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.VisualBasic.CodeStyle" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.CodeStyle.UnitTestUtilities" />
-    <InternalsVisibleTo Include="Microsoft.CodeAnalysis.CSharp.CodeStyle.UnitTests" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.VisualBasic.CodeStyle.UnitTests" />
```
Comment (mavasani) [inline src/CodeStyle/Core/Analyzers/Microsoft.CodeAnalysis.CodeStyle.csproj:29]: `Microsoft.CodeAnalysis.VisualBasic.CodeStyle.UnitTests` has tests?

File: src/EditorFeatures/DiagnosticsTestUtilities/Microsoft.CodeAnalysis.EditorFeatures.DiagnosticsTests.Utilities.csproj [hunk 5]
```diff
@@ -49,10 +49,4 @@
     <InternalsVisibleTo Include="Microsoft.VisualStudio.IntegrationTest.Utilities" />
     <InternalsVisibleTo Include="Microsoft.VisualStudio.LanguageServices.Test.Utilities2" />
   </ItemGroup>
-  <ItemGroup>
-    <!-- TODO: Remove the below IVTs to CodeStyle Unit test projects once all analyzer/code fix tests are switched to Microsoft.CodeAnalysis.Testing -->
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/EditorFeatures/DiagnosticsTestUtilities/Microsoft.CodeAnalysis.EditorFeatures.DiagnosticsTests.Utilities.csproj:53]: @mavasani The comment should have been referring to tests in `CodeStyle`, not all tests.

In this PR, I removed the `CodeStyle` test projects altogether. All tests should be in the shared `Analyzers` layer.

File: src/CodeStyle/Core/Analyzers/Microsoft.CodeAnalysis.CodeStyle.csproj (hunk 4 above)
Comment (Youssef1313) [reply src/CodeStyle/Core/Analyzers/Microsoft.CodeAnalysis.CodeStyle.csproj:29]: Project is completely deleted. I'll clean this up.

File: src/CodeStyle/CSharp/Tests/AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest_OptionHelpers.cs [hunk 6]
```diff
@@ -1,54 +0,0 @@
...
-
-#nullable disable
-
-using Microsoft.CodeAnalysis.CodeStyle;
-using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
-using Microsoft.CodeAnalysis.Options;
-
-namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics
-{
-    public abstract partial class AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/CodeStyle/CSharp/Tests/AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest_OptionHelpers.cs:13]: @mavasani In `CodeStyle` layer, yes it seems it was unused. And formatting tests are now using the new testing library.

**Comment by CyrusNajmabadi** [comment]
This is AWESOME.  Thanks so much @Youssef1313 !

**Review by mavasani — APPROVED** [review]
Thank you @Youssef1313

**Comment by CyrusNajmabadi** [comment]
Done!

**Comment by Youssef1313** [comment]
@CyrusNajmabadi I fixed merge conflicts so auto merge got disabled. Can you enable it again? Thanks!

**Comment by CyrusNajmabadi** [comment]
Done!

File: src/CodeStyle/CSharp/Tests/Microsoft.CodeAnalysis.CSharp.CodeStyle.UnitTests.csproj [hunk 7]
```diff
@@ -1,60 +0,0 @@
...
-    <ProjectReference Include="..\..\..\Features\LanguageServer\Protocol\Microsoft.CodeAnalysis.LanguageServer.Protocol.csproj" />
-    <ProjectReference Include="..\..\..\Workspaces\Remote\Core\Microsoft.CodeAnalysis.Remote.Workspaces.csproj" />
-    <ProjectReference Include="..\..\..\Workspaces\Remote\ServiceHub\Microsoft.CodeAnalysis.Remote.ServiceHub.csproj" />
-    <ProjectReference Include="..\..\..\Compilers\Test\Resources\Core\Microsoft.CodeAnalysis.Compiler.Test.Resources.csproj" />
-    <ProjectReference Include="..\..\..\EditorFeatures\TestUtilities\Microsoft.CodeAnalysis.EditorFeatures.Test.Utilities.csproj" />
-    <ProjectReference Include="..\..\..\Test\PdbUtilities\Roslyn.Test.PdbUtilities.csproj" />
-    <ProjectReference Include="..\..\..\Compilers\Test\Core\Microsoft.CodeAnalysis.Test.Utilities.csproj" />
-    <ProjectReference Include="..\..\..\Workspaces\CoreTestUtilities\Microsoft.CodeAnalysis.Workspaces.Test.Utilities.csproj" />
-  </ItemGroup>
-  <Import Project="..\..\..\Analyzers\CSharp\Tests\CSharpAnalyzers.UnitTests.projitems" Label="Shared" />
```
Comment (mavasani) [inline src/CodeStyle/CSharp/Tests/Microsoft.CodeAnalysis.CSharp.CodeStyle.UnitTests.csproj:59]: @Youssef1313 My bad that I did not notice that your change actually deletes a bunch of test projects. C# and VB CodeStyle unit test projects test the analyzers that ship as part of respective CodeStyle NuGet package. I am going to file an issue to restore these projects back.

Comment (Youssef1313) [reply src/CodeStyle/CSharp/Tests/Microsoft.CodeAnalysis.CSharp.CodeStyle.UnitTests.csproj:59]: Oops. Yeah sorry for that.

**Pull request merged**